### DISCUSSION TITLE
USB驱动，在VS环境下测试的疑问

### DISCUSSION THREAD

**Discussion by jinsonli** [peng-zhihui/ElectronBot#96] [Q&A]
我自己尝试下了用libusb库，在linux下能正常通讯， 在windows下能list device， 用PID 和VID 打开device 总是NULL，怀疑跟权限有关。
目前资料里提供的的USBInterface.dll在我这边的VS版本下跑不起来。如果能提供源码的话，希望先通过源码集成下，看看坑在哪里，弄完我分享下细节。
感谢感谢。

**Comment by peng-zhihui** [comment]
正常测试和二次开发不涉及USBInterface源码的，不过我回去可以发你。你确认一下win的驱动是否正常安装（需要禁用强制签名，参考README）

**Reply by coolflyreg** [reply]
> 正常测试和二次开发不涉及USBInterface源码的，不过我回去可以发你。你确认一下win的驱动是否正常安装（需要禁用强制签名，参考README）

如果稚晖君要开源USBInterfaceDll的代码，我可以把我改的osx相关的编译修改，提交PR，这样可以多平台运行

**Reply by peng-zhihui** [reply]
可以的

**Comment by jinsonli** [comment]
感谢提醒，搞定了。 我这个 libusb 通讯程序 在WINDOWS下 open不成功的原因确实是因为驱动的问题。 
之前调试STM32 VCP 时，装成了 Usbser. ， 现在用 Zadig把驱动换成 libusb-win32 或者 WinUsb就可以。
晚点我调试好了整个DEMO给大家吧，老是麻烦稚晖君鸭梨太大了。
![9bee46f840c49b6fea8d888199f4877](https://user-images.githubusercontent.com/26651232/159274057-ef4cb014-b1dd-42d4-a329-52b6a04493f8.png)
![7231e71ebee3fd99607271e886ba92f](https://user-images.githubusercontent.com/26651232/159274072-c4f50cbd-8a04-41ef-87f3-8e8df060e750.png)

**Reply by coolflyreg** [reply]
厉害了

**Comment by coolflyreg** [comment]
> 我自己尝试下了用libusb库，在linux下能正常通讯， 在windows下能list device， 用PID 和VID 打开device 总是NULL，怀疑跟权限有关。 目前资料里提供的的USBInterface.dll在我这边的VS版本下跑不起来。如果能提供源码的话，希望先通过源码集成下，看看坑在哪里，弄完我分享下细节。 感谢感谢。

请问，在Linux下是如何正常通讯的呢？我用树莓派尝试的，但是虽然能连上，但没有数据传输
<img width="622" alt="image" src="https://user-images.githubusercontent.com/4011564/163230498-2ab0b5ea-1474-45ac-a61a-acca420cad27.png">

**Reply by erikaemma** [reply]
请问，有PR或者DEMO分享吗，大佬？
